src/components/stepper/stepper.tsx

```diff
@@ -1,47 +1,139 @@
-                type="button"
-                tabIndex={-1}
-                data-automation-id="STEPPER_DECREMENT"
-                className={`${buttonStyles.root} control down`}
-                onClick={onDown}
-                disabled={disableDown}
+export interface State {
+    dragged: boolean;
+}
+
+export interface Modifiers {
+    altKey?: boolean;
+    ctrlKey?: boolean;
+    shiftKey?: boolean;
+}
+
+interface DragPoint {
+    clientX: number;
+    clientY: number;
+}
+
+const DEFAULTS = {
+    dragStep: 10,
+    disableUp: false,
+    disableDown: false
+};
+
+@SBComponent(styles)
+export class Stepper extends React.Component<StepperProps, State> {
+    public static defaultProps = {
+        disableUp: DEFAULTS.disableUp,
+        disableDown: DEFAULTS.disableDown
+    };
+
+    public state: State = {dragged: false};
+
+    private dragRefPoint: DragPoint = {
+        clientX: 0,
+        clientY: 0
+    };
+
+    public render() {
+        const {dragged} = this.state;
+        const {
+            onUp,
+            onDown,
+            disableUp,
+            disableDown,
+            dragStep,
+            ...props
+        } = this.props;
+
+        return (
+            <div
+                {...props}
+                onMouseDown={this.handleDragStart}
             >
-                <ChevronDownIcon className="control-icon"/>
-            </button>
-        </div>
-    ),
-    styles
-);
+                <button
+                    type="button"
+                    tabIndex={-1}
+                    data-automation-id="STEPPER_INCREMENT"
+                    className={`${buttonStyles.root} control up`}
+                    onClick={this.handlerClickUp}
+                    disabled={disableUp}
+                >
+                    <ChevronUpIcon className="control-icon" />
+                </button>
+                <button
+                    type="button"
+                    tabIndex={-1}
+                    data-automation-id="STEPPER_DECREMENT"
+                    className={`${buttonStyles.root} control down`}
+                    onClick={this.handlerClickDown}
+                    disabled={disableDown}
+                >
+                    <ChevronDownIcon className="control-icon"/>
+                </button>
+                <GlobalEvent
+                    mousemove={dragged ? this.handleDrag : undefined}
+                    mouseup={this.handleDragStop}
+                />
+            </div>
+        );
+    }
+
+    private handlerClickUp: React.MouseEventHandler<HTMLButtonElement> =
+        ({altKey, ctrlKey, shiftKey}) => this.props.onUp({altKey, ctrlKey, shiftKey})
+
+    private handlerClickDown: React.MouseEventHandler<HTMLButtonElement> =
+        ({altKey, ctrlKey, shiftKey}) => this.props.onDown({altKey, ctrlKey, shiftKey})
+
+    private handleDragStart: React.MouseEventHandler<HTMLDivElement> =
+        ({clientX, clientY}) => {
+            this.setState({dragged: true});
```

Comment:
`dragged` better to keep in instance `dragRefPoint`. No needs to render when this flag is changed